idealingua/idealingua-runtime-rpc-scala/src/main/scala/com/github/pshirshov/izumi/idealingua/runtime/rpc/IRTCirceMarshaller.scala

```diff
@@ -4,16 +4,18 @@ import io.circe.{DecodingFailure, Json}
 
 import scala.language.higherKinds
 
-trait IRTCirceMarshaller[R[_, _]] extends IRTResultTransZio[R] {
+abstract class IRTCirceMarshaller[R[_, _] : IRTResult] {
+  val R: IRTResult[R] = implicitly[IRTResult[R]]
+
   def encodeRequest: PartialFunction[IRTReqBody, Json]
 
   def encodeResponse: PartialFunction[IRTResBody, Json]
 
-  def decodeRequest: PartialFunction[IRTJsonBody, Just[IRTReqBody]]
+  def decodeRequest: PartialFunction[IRTJsonBody, R.Just[IRTReqBody]]
```

Comment:
We could also remove parameter from IRTCirceMarshallers by accepting parameter in function instead:

```
def decodeResponse[R[_. _]: IRTResult]: PartialFunction[IRTJsonBody, R.Just[IRTResBody]]
```